src/results.rs

```diff
@@ -11,6 +12,23 @@ use std::str::FromStr;
 use toolchain::Toolchain;
 use util;
 
+pub trait ExperimentResultDB {
```

Comment:
The names here aren't the greatest. I'd be happy for suggestions on better names.